toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1019,8 +1019,22 @@ func (g *PkgGraph) AddMetaNode(from []*PkgNode, to []*PkgNode) (metaNode *PkgNod
 	return
 }
 
-// AddGoalNode adds a goal node to the graph which links to existing nodes. An empty package list will add an edge to all nodes
+// AddGoalNodeWithLayers adds a goal node to the graph which links to existing nodes. An empty package list will add an edge to all nodes
+//   - goalName: The name of the goal node to add
+//   - packages: A list of packages to add to link the goal node to. If empty, all nodes will be added to the goal node
+//   - strict: If true, the goal node will fail if any of the packages are not found
 func (g *PkgGraph) AddGoalNode(goalName string, packages []*pkgjson.PackageVer, strict bool) (goalNode *PkgNode, err error) {
+	return g.AddGoalNodeWithExtraLayers(goalName, packages, strict, 0)
+}
+
+// AddGoalNodeWithExtraLayers adds a goal node to the graph which links to existing nodes. An empty package list will add an edge to all nodes
+//   - goalName: The name of the goal node to add
+//   - packages: A list of packages to add to link the goal node to. If empty, all nodes will be added to the goal node
+//   - strict: If true, the goal node will fail if any of the packages are not found
+//   - extraLayers: The number of levels to expand the goal node. Each level will add one more layer of packages beyond
+//     the goal node. For example, if the goal node is "foo" and extraLevels is 1, the goal node will link to all nodes
+//     which depend on "foo" as well as "foo" itself (Specifically run nodes, all other nodes are stepped over)
+func (g *PkgGraph) AddGoalNodeWithExtraLayers(goalName string, packages []*pkgjson.PackageVer, strict bool, extraLayers int) (goalNode *PkgNode, err error) {
```

Comment:
Question about a scenario we want to support.

Context:
- We've just published all latest packages to PMC.
- We're now fixing a critical CVE in `foo`.

The delta build will now run `make build-packages CONFIG_FILE= DELTA_FETCH=y EXTRA_BUILD_LAYERS=1`. It is not expected to try and figure out, which packages changed, thus no `PACKAGE_BUILD_LIST` argument. Will this implementation allow us to build `foo` and its closest dependents?

Reply:
This implementation will have no meaningful impact in this scenario. We will be using the automatic ALL goal node which points to all packages, so adding addional layers will have no affect here. A follow on feature will be needed to better optimize this use case (by limiting the cascading rebuilds).

Reply:
(We did this in https://github.com/microsoft/CBL-Mariner/pull/5993)